src/components/Button/index.tsx

```diff
@@ -16,195 +15,155 @@ import themeColors from '@styles/themes/default';
 
     /** Should enable the haptic feedback? */
-    shouldEnableHapticFeedback: PropTypes.bool,
+    shouldEnableHapticFeedback?: boolean;
 
     /** Id to use for this button */
-    id: PropTypes.string,
+    id?: string;
 
     /** Accessibility label for the component */
-    accessibilityLabel: PropTypes.string,
+    accessibilityLabel?: string;
 
     /** A ref to forward the button */
-    forwardedRef: refPropTypes,
-};
-
-const defaultProps = {
-    allowBubble: false,
-    text: '',
-    shouldShowRightIcon: false,
-    icon: null,
-    iconRight: Expensicons.ArrowRight,
-    iconFill: themeColors.textLight,
-    iconStyles: [],
-    iconRightStyles: [],
-    isLoading: false,
-    isDisabled: false,
-    small: false,
-    large: false,
-    medium: false,
-    onPress: () => {},
-    onLongPress: () => {},
-    onPressIn: () => {},
-    onPressOut: () => {},
-    onMouseDown: undefined,
-    pressOnEnter: false,
-    enterKeyEventListenerPriority: 0,
-    style: [],
-    innerStyles: [],
-    textStyles: [],
-    shouldUseDefaultHover: true,
-    success: false,
-    danger: false,
-    children: null,
-    shouldRemoveRightBorderRadius: false,
-    shouldRemoveLeftBorderRadius: false,
-    shouldEnableHapticFeedback: false,
-    id: '',
-    accessibilityLabel: '',
-    forwardedRef: undefined,
+    forwardedRef?: React.ForwardedRef<View>;
 };
 
 function Button({
-    allowBubble,
-    text,
-    shouldShowRightIcon,
-
-    icon,
-    iconRight,
-    iconFill,
-    iconStyles,
-    iconRightStyles,
-
-    small,
-    large,
-    medium,
-
-    isLoading,
-    isDisabled,
-
-    onPress,
-    onLongPress,
-    onPressIn,
-    onPressOut,
-    onMouseDown,
-
-    pressOnEnter,
-    enterKeyEventListenerPriority,
-
-    style,
-    innerStyles,
-    textStyles,
-
-    shouldUseDefaultHover,
-    success,
-    danger,
-    children,
-
-    shouldRemoveRightBorderRadius,
-    shouldRemoveLeftBorderRadius,
-    shouldEnableHapticFeedback,
-
-    id,
-    accessibilityLabel,
-    forwardedRef,
-}) {
+    allowBubble = false,
+    text = '',
+    shouldShowRightIcon = false,
+
+    icon = null,
+    iconRight = Expensicons.ArrowRight,
+    iconFill = themeColors.textLight,
+    iconStyles = [],
+    iconRightStyles = [],
+
+    small = false,
+    large = false,
+    medium = false,
+
+    isLoading = false,
+    isDisabled = false,
+
+    onPress = () => {},
+    onLongPress = () => {},
+    onPressIn = () => {},
+    onPressOut = () => {},
+    onMouseDown = undefined,
+
+    pressOnEnter = false,
+    enterKeyEventListenerPriority = 0,
+
+    style = [],
+    innerStyles = [],
+    textStyles = [],
+
+    shouldUseDefaultHover = true,
+    success = false,
+    danger = false,
+    children = null,
+
+    shouldRemoveRightBorderRadius = false,
+    shouldRemoveLeftBorderRadius = false,
+    shouldEnableHapticFeedback = false,
+
+    id = '',
+    accessibilityLabel = '',
+    forwardedRef = undefined,
+}: ButtonProps) {
```

Comment:
Get rid of `forwardedRef` in props, put it as a second prop, this way you can get rid of `ButtonWithRef`
```suggestion
}: ButtonProps, ref: React.ForwardedRef<View>) {
```